### ISSUE TITLE
Suggestion: static panel, sliding menu.

### ISSUE THREAD

**Issue by mauron85** [Mango/slideout#140]
Current implementation is that menu is static and panel is moved to side to reveal menu by translate3d.

However, most menu implementations (tinder, facebook) actually do it in reverse manner, so
panel is static and menu is revealing.

Can this be implemented?

**Comment by wesleyfsmith** [comment]
Yes, this is also how I would prefer to use this library as well. Would be good to know if anyone knows how to get it functioning in this manner.

**Comment by AaronLayton** [comment]
@pazguille I too would like to pull out the Menu as a Panel - Would this just be as easy as swapping around what we class as the Menu and the Panel?

**Comment by pazguille** [comment]
+1 Yes, it should be configurable to support both cases well.

**Comment by pazguille** [comment]
@AaronLayton It is not so simple. I'll work in this feature.

**Comment by AaronLayton** [comment]
I got it working the other night how I figured it should work. I will see
if I can post you a pull request ;-)

**Comment by nolandg** [comment]
Did anyone get this working? I don't seem to see the pull request...

**Comment by stevenmunro** [comment]
I began working on this until I saw this issue. I managed to swap out instances of panel with menu in the JS file, however I'm having problems with the translate properties between transitions.

Does anyone have any code they want to share?

**Comment by slzr** [comment]
:+1:

**Comment by slzr** [comment]
+1

**Comment by nstanard** [comment]
Would also like to see this happen.

**Comment by groomain** [comment]
+1 :)

**Comment by aguilera51284** [comment]
+1

**Comment by AaronLayton** [comment]
Sorry it took a while everyone but here is the modifications I made to get this working. Hopefully my PR can be used as a starting point to get a better implementation, but if nothing comes of this then you can use the version here
https://github.com/AaronLayton/slideout

**Comment by AaronLayton** [comment]
Any news on this PR?

**Comment by erichonkanen** [comment]
+1 any movement?

**Comment by pazguille** [comment]
Hi @erichonkanen! 

I have been working on this feature and other improvements here:

Branch:
https://github.com/mango/slideout/tree/new-features

Demo:
https://slideout.js.org/new-features

Soon it going to be merged into master and I'll publish a new version 2.0.0

**Comment by AaronLayton** [comment]
@pazguille great work! I like how your not changing the display of the `slideout-menu` anymore - that was always a perf issue for me. Having to draw the menu just before we slide it out was causing some hickups on slower devices.

Another perf issue for me is adding the `slideout-open` class to the `<html>` tag. Any chance this could be changed to a visible class just being added to a target element? Either add `slideout-dimmer` as default or something else we can specify?

**Comment by pazguille** [comment]
@AaronLayton Yes, I think there are no problem to do that. I will work on your comments. Thanks!

**Comment by JacobDB** [comment]
@pazguille Any progress on integrating this in to master? Would love to use the overlay variant!
